src/views/Users/CreateUser/CreateUser.tsx

```diff
@@ -0,0 +1,168 @@
+import { Box, Button, Stack, Typography } from '@mui/material'
+import React from 'react'
+import Input from '../../../components/Input'
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import { phoneRegExp } from '../../../helpers/regex';
+import { addUser } from '../../../redux/actions/users';
+import { useDispatch } from 'react-redux';
+
+const requiredMessage = 'This is a required field';
+
+const createUserValidation = Yup.object().shape({
+  name: Yup.string().required(requiredMessage),
+  username: Yup.string().required(requiredMessage),
+  phone: Yup.string().matches(phoneRegExp, 'Phone number is not valid').required(requiredMessage),
+  website: Yup.string().url('Please enter a valid URL').required(requiredMessage),
+  email: Yup.string().email('Please enter a valid email').required(requiredMessage),
+  street: Yup.string().required(requiredMessage),
+  suite: Yup.string().required(requiredMessage),
+  city: Yup.string().required(requiredMessage),
+  zipcode: Yup.string().required(requiredMessage),
+});
+
+const fieldNames = {
+  NAME: 'name',
+  USERNAME: 'username',
+  PHONE: 'phone',
+  WEBSITE: 'website',
+  EMAIL: 'email',
+  STREET: 'street',
+  SUITE: 'suite',
+  CITY: 'city',
+  ZIPCODE: 'zipcode',
+}
+
+interface FormValues  {
+  name: string; 
+  username: string;
+  phone: string;
+  website: string;
+  email: string;
+  street: string;
+  suite: string;
+  city: string;
+  zipcode: string;
+}
+
+const CreateUser = () => {
+  const initialValues : FormValues = {
+      name: '',
+      username: '',
+      phone: '',
+      website: '',
+      email: '',
+      street: '',
+      suite: '',
+      city: '',
+      zipcode: ''
+  }
+  const { handleChange, handleBlur, values, handleSubmit, errors } = useFormik({
+    initialValues: initialValues,
+    validateOnBlur: false,
+    validateOnChange: false,
+    validationSchema: createUserValidation,
+    onSubmit: (val) => {
+      console.log(val);
+      addUser(val);
+    },
+  });
+
+  return (
+    <Box sx={{
+      display: 'flex',
+      flexWrap: 'wrap',
+      flexDirection: 'column',
+      justifyContent: 'center',
+      alignItems: 'center'
+    }}>
+      <Typography mb={5} variant="h2">Create User</Typography>
+
+      <Stack spacing={2} component='form' onSubmit={handleSubmit} >
+        <Input onChange={handleChange}
+          onBlur={handleBlur}
+          id={fieldNames.NAME}
+          label="Name"
+          errorText={errors.name}
+          error={errors.name ? true : false}
+          value={values.name} 
+          />
```

Comment:
This is being re-used multiple times in this file. Can you loop over the list of fields and use this component only once ?

Reply:
Done